`vue-tsc` throws error when importing `vite-ssg/single-page`

When importing from `vite-ssg/single-page` in a TypeScript, `vue-tsc` will throw an error.

This is due to the script `scripts/copy-single-page-dts.ts`, which blindly copies the type definitions in the dist root, so it can be found by `tsc` at `vite-ssg/single-page.d.ts`.

What causes the error is a relative import in `single-page.d.ts`, which is incorrect after moving it one folder up:
[`single-page.ts`](https://github.com/antfu/vite-ssg/blob/main/src/client/single-page.ts#L5-L7):
```ts
// ...
import type { ViteSSGClientOptions, ViteSSGContext } from '../types'
import { deserializeState } from '../utils/state'
import { documentReady } from '../utils/document-ready'
// ...
```
which compiles e. g. to `single-page.d.ts`:
```js
import { Component } from 'vue';
import { V as ViteSSGContext, a as ViteSSGClientOptions } from '../types-afd5a7d9';
export { R as RouterOptions, a as ViteSSGClientOptions, V as ViteSSGContext, b as ViteSSGOptions } from '../types-afd5a7d9';
import 'vue-router';
import '@vueuse/head';
import 'critters';

declare function ViteSSG(App: Component, fn?: (context: ViteSSGContext<false>) => Promise<void> | void, options?: ViteSSGClientOptions): (client?: boolean) => Promise<ViteSSGContext<false>>;

export { ViteSSG };
```

When running `vue-tsc`, I get this error:
```bash
node_modules/vite-ssg/single-page.d.ts:2:64 - error TS2307: Cannot find module '../types-afd5a7d9' or its corresponding type declarations.

2 import { V as ViteSSGContext, a as ViteSSGClientOptions } from '../types-afd5a7d9';
                                                                 ~~~~~~~~~~~~~~~~~~~

node_modules/vite-ssg/single-page.d.ts:3:105 - error TS2307: Cannot find module '../types-afd5a7d9' or its corresponding type declarations.

3 export { R as RouterOptions, a as ViteSSGClientOptions, V as ViteSSGContext, b as ViteSSGOptions } from '../types-afd5a7d9';
                                                                                                          ~~~~~~~~~~~~~~~~~~~
Found 2 errors.
```

If I go in manually and fix the paths to ` './dist/types-afd5a7d9'`, it compiles again.

--- a/package.json
+++ b/package.json
@@ -33,6 +33,11 @@
   "main": "dist/index.cjs",
   "module": "dist/index.mjs",
   "types": "dist/index.d.ts",
+  "typesVersions": {
+    "*": {
+      "single-page": ["dist/client/single-page.d.ts"]
+    }
+  },
   "files": [
     "dist",
     "bin",
@@ -45,7 +50,6 @@
     "dev": "tsup --watch src",
     "lint": "eslint --ext .ts,.js .",
     "copy-readme-files": "esno scripts/copy-readme-files.ts",
-    "copy-single-page-dts": "esno scripts/copy-single-page-dts.ts",
     "example:dev": "npm run copy-readme-files && npm -C examples/multiple-pages run dev",
     "example:build": "npm run copy-readme-files && npm -C examples/multiple-pages run build",
     "example:serve": "npm run copy-readme-files && npm -C examples/multiple-pages run serve",
@@ -55,7 +59,7 @@
     "example:single:dev": "npm -C examples/single-page run dev",
     "example:single:build": "npm -C examples/single-page run build",
     "example:single:serve": "npm -C examples/single-page run serve",
-    "build": "unbuild && npm run copy-single-page-dts",
+    "build": "unbuild",
     "prepublishOnly": "npm run build",
     "release": "npx bumpp --push --tag --commit"
   },

--- a/scripts/copy-single-page-dts.ts
+++ b/scripts/copy-single-page-dts.ts
@@ -1,3 +1,0 @@
-import { copyFileSync } from 'fs'
-
-copyFileSync('dist/client/single-page.d.ts', 'single-page.d.ts')

